pkg/utils/has/controller.go

```diff
@@ -268,6 +274,17 @@ func (h *SuiteController) ComponentDeleted(component *appservice.Component) wait
 	}
 }
 
+func (h *SuiteController) CheckForImageAnnotation(component *appservice.Component) wait.ConditionFunc {
+	return func() (bool, error) {
+		componentCR, err := h.GetHasComponent(component.Name, component.Namespace)
+		if err != nil {
+			return false, nil
+		}
+		annotations := componentCR.GetAnnotations()
+		return build.IsImageAnnotationPresent(annotations) && build.ImageAnnotationGenerateValueIsNotFailed(annotations), nil
```

Comment:
in case that the annotation is not present then this check is irrelevant and should not timeout.

Reply:
Since the `image.redhat.com/image` annotation takes sometime to appear after creating the component. So I was retrying till 1 minute for this annotation to present and timeout if not exists after 1 minute. Do you think we should change the logic?